dist/iconpicker-1.5.0.js

```diff
@@ -1,439 +1,432 @@
-
-                        // callback on
-                        if (theCallback) {
-                            theCallback();
-                        }
-                        // callback off
-
-                        removeIpElement(400);
-                    }, false);
-                }
-                // add listeners each off
+            var noResultElm = '<p class="ip-no-results-found">' + noResultsText + '</p>';
+            tempResults.innerHTML = noResultElm;
+          }
+          // no results found off
 
-            }
-            // each icon click listener off
+          // append temp container to results area
+          searchResultArea.appendChild(tempResults);
 
-            // first icons listeners
-            eachIconEventListener('first');
+          // event listener for each temp icon
+          eachIconEventListener('search');
 
         }
-        // IconPicker: Append Library to Body off
-    },
+        // show first icons
+        else {
+          firstIconsArea.style.display = 'block';
+        }
+
+      }, false);
+      // search input keyup listener off
+
+
+      // each icon click listener on
+      function eachIconEventListener(firstOrSearch) {
+
+        var inputElm = document.querySelectorAll(inputElement);
+
+        // define icons on
+        var eachIconElm;
+        if (firstOrSearch === 'first') { // first
+          eachIconElm = document.getElementById(IconPickerModal.id).getElementsByClassName('first-icon');
+        } else if (firstOrSearch === 'search') { // search
+          eachIconElm = document.getElementById(IconPickerModal.id).getElementsByClassName('search-icon');
+        }
+        // define icons off
+
+        // add listeners each on
+        var wrapElem = document.querySelector('.ip-icons-content');
+        var config = {
+          attributes: true,
+          childList: true,
+          subtree: true
+        };
+
+        var callback = function(mutationsList, observer) {
+          attachClickEventForEachIcon(eachIconElm, inputElm);
+        };
+
+        var observer = new MutationObserver(callback);
+        observer.observe(wrapElem, config);
+
+        attachClickEventForEachIcon(eachIconElm, inputElm);
+
+        // add listeners each off
+
+      }
+
+      function attachClickEventForEachIcon(eachIconElm, inputElm) {
+
+        for (var i = 0; i < eachIconElm.length; i++) {
+          var singleIconElm = eachIconElm[i];
+          singleIconElm.addEventListener('click', function(e) {
+            e.preventDefault();
+            var iconClassName = this.dataset.class;
+            for (var i = 0; i < inputElm.length; i++) {
+
+              if (inputElm[i].tagName === 'INPUT') {
+                inputElm[i].value = iconClassName;
+              } else {
+                inputElm[i].innerHTML = iconClassName;
+              }
+              $("#macroIconPreview").html("<i class='" + iconClassName + " fa-2x m-2'></i>");
```

Comment:
The whole code uses rather plain js, and since i cannot use jq in place where i use this plugin then i already wrote working vanilla:
````
                            let element = document.getElementById('macroIconPreview');
                            if( null !== element ){
                                element.innerHTML = "<i class='" + iconClassName + " fa-2x m-2'></i>";
                            }
````